lms/djangoapps/courseware/views/index.py

```diff
@@ -213,47 +199,7 @@ def _redirect_if_needed_to_pay_for_course(self):
                 self.real_user,
                 unicode(self.course_key),
             )
-            raise Redirect(reverse('dashboard'))
-
-    def _redirect_if_needed_to_register(self):
```

Comment:
This logic is already encapsulated in ``get_course_with_access``, so the only fix I made was to pass ``check_if_enrolled=True`` when using it.